lib/checkother.cpp

```diff
@@ -2852,8 +2852,16 @@ void CheckOther::checkIncompleteArrayFill()
 
     for (const Scope * scope : symbolDatabase->functionScopes) {
         for (const Token* tok = scope->bodyStart->next(); tok != scope->bodyEnd; tok = tok->next()) {
-            if (Token::Match(tok, "memset|memcpy|memmove ( %var% ,") && Token::Match(tok->linkAt(1)->tokAt(-2), ", %num% )")) {
-                const Variable *var = tok->tokAt(2)->variable();
+            if (Token::Match(tok, "memset|memcpy|memmove ( ") && Token::Match(tok->linkAt(1)->tokAt(-2), ", %num% )")) {
+                const Token* tok2 = tok->tokAt(2);
```

Comment:
As far as I see we would still not warn about:

    struct B { int a[5]; };
    struct C { struct B b; };

    void f(struct C *c) {
        memset(c[1].b.a, 0, 5);
    }

what do you think about making this checker more generic?

using a size value `5` that is not a multiple of `sizeof(int)` makes it even more suspicious in this case.

I don't suggest that you do it in this PR but if we warn for arrays we should probably warn for unpacked structs also somehow. For arrays I can imagine that memset could be used partially in some use cases but to me it's harder to imagine that for structs.